paparazzi/src/main/java/app/cash/paparazzi/PaparazziSdk.kt

```diff
@@ -0,0 +1,653 @@
+import com.android.layoutlib.bridge.BridgeRenderSession
+import com.android.layoutlib.bridge.impl.RenderAction
+import com.android.layoutlib.bridge.impl.RenderSessionImpl
+import com.android.resources.ScreenOrientation
+import com.android.resources.ScreenRound
+import com.android.tools.idea.validator.LayoutValidator
+import com.android.tools.idea.validator.ValidatorData.Level
+import com.android.tools.idea.validator.ValidatorData.Policy
+import com.android.tools.idea.validator.ValidatorData.Type
+import net.bytebuddy.agent.ByteBuddyAgent
+import java.awt.geom.Ellipse2D
+import java.awt.image.BufferedImage
+import java.util.EnumSet
+import java.util.concurrent.TimeUnit
+import kotlinx.coroutines.android.asCoroutineDispatcher
+
+@OptIn(ExperimentalComposeUiApi::class, InternalComposeUiApi::class)
+public class PaparazziSdk @JvmOverloads constructor(
+  private val environment: Environment = detectEnvironment(),
+  private val deviceConfig: DeviceConfig = DeviceConfig.NEXUS_5,
+  private val theme: String = "android:Theme.Material.NoActionBar.Fullscreen",
+  private val renderingMode: RenderingMode = RenderingMode.NORMAL,
+  private val appCompatEnabled: Boolean = true,
+  private val renderExtensions: Set<RenderExtension> = setOf(),
+  private val supportsRtl: Boolean = false,
+  private val showSystemUi: Boolean = false,
+  private val validateAccessibility: Boolean = false
+) {
+  private val logger = PaparazziLogger()
+  private lateinit var renderSession: RenderSessionImpl
+  private lateinit var bridgeRenderSession: RenderSession
+  private var frameHandler: FrameHandler? = null
+
+  public val layoutInflater: LayoutInflater
+    get() = RenderAction.getCurrentContext().getSystemService("layout_inflater") as BridgeInflater
+
+  public val resources: Resources
+    get() = RenderAction.getCurrentContext().resources
+
+  public val context: Context
+    get() = RenderAction.getCurrentContext()
+
+  private val contentRoot = """
+        |<?xml version="1.0" encoding="utf-8"?>
+        |<${if (hasComposeRuntime) "app.cash.paparazzi.internal.ComposeViewAdapter" else "FrameLayout"}
+        |     xmlns:android="http://schemas.android.com/apk/res/android"
+        |              android:layout_width="${if (renderingMode.horizAction == RenderingMode.SizeAction.SHRINK) "wrap_content" else "match_parent"}"
+        |              android:layout_height="${if (renderingMode.vertAction == RenderingMode.SizeAction.SHRINK) "wrap_content" else "match_parent"}"/>
+  """.trimMargin()
+
+  public fun setup() {
+    if (!isInitialized) {
+      registerFontLookupInterceptionIfResourceCompatDetected()
+      registerViewEditModeInterception()
+      registerMatrixMultiplyInterception()
+      registerChoreographerDelegateInterception()
+      registerServiceManagerInterception()
+      registerIInputMethodManagerInterception()
+
+      ByteBuddyAgent.install()
+      InterceptorRegistrar.registerMethodInterceptors()
+    }
+
+    val layoutlibCallback =
```

Comment:
```suggestion
    }
  }
  
  public fun prepare() {
    val layoutlibCallback =
```

deliberate to decouple the interceptor setup from the layoutlib bootstrapping.